File: TripleX/TripleX.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <stdbool.h>

bool bCompleteIntroduction = false;

void PrintIntroduction()
{
  printf("Hello student, welcome to the math quizzes!\n\n");
}

int AddVariables(int GuessA, int GuessB, int GuessC)
{
  return GuessA + GuessB + GuessC; 
}

int MultiplyVariables(int GuessA, int GuessB, int GuessC)
{
  return GuessA * GuessB * GuessC; 
}

bool PlayGame(int Difficulty)
{
  const int DifficultyScaling = pow(3,Difficulty) + 1;
  if (Difficulty <= 1)
  {
      PrintIntroduction();
  }
  const int CodeA = rand() % DifficultyScaling;
  const int CodeB = rand() % DifficultyScaling;
  const int CodeC = rand() % DifficultyScaling;
  const int CodeSum = CodeA + CodeB + CodeC;
  const int CodeProduct = CodeA * CodeB * CodeC;
  int GuessA = 0, GuessB = 0, GuessC = 0;
  int GuessSum = 0, GuessProduct = 0;
  printf("These numbers add up to: %d", CodeSum);
  printf(" and multiply to: %d\n", CodeProduct);
  scanf("%d", &GuessA); 
  scanf("%d", &GuessB); 
  scanf("%d", &GuessC); 
  GuessSum = AddVariables(GuessA, GuessB, GuessC);
  GuessProduct = MultiplyVariables(GuessA, GuessB, GuessC);
  return (GuessSum == CodeSum && GuessProduct == CodeProduct);
}

int main()
{
  const int MaxDifficulty = 6;
  int Difficulty = 1;
  while (Difficulty <= MaxDifficulty)
  {
    bool bCompletedGame = PlayGame(Difficulty);
    if (bCompletedGame == true)
    {
      printf("Nice work! Let's get a little tougher.\n\n");
      ++Difficulty;
    }
    else
    {
      printf("No, no, this won't do. Try a different one.\n\n");
    }
  }
  printf("Quiz Completed.");
  return 0;
}

File: TripleX/TripleX.java
import java.util.*;

class Main {
  public static boolean PlayGame(int Difficulty)
  {
    Random RNG = new Random();
    int DifficultyScaling = 3 * Difficulty + 1;
    int CodeA = RNG.nextInt() % DifficultyScaling;
    if (CodeA < 0)
    {
      CodeA = CodeA + (CodeA*-2);
    }
    int CodeB = RNG.nextInt() % DifficultyScaling;
    if (CodeB < 0)
    {
      CodeB = CodeB + (CodeB*-2);
    }
    int CodeC = RNG.nextInt() % DifficultyScaling;
    if (CodeC < 0)
    {
      CodeC = CodeC + (CodeC*-2);
    }
    int CodeSum = CodeA + CodeB + CodeC;
    int CodeProduct = CodeA * CodeB *CodeC;
    Scanner input = new Scanner(System.in);
    System.out.println("What are numbers that add up to:" + CodeSum + " and multiply up to: " + CodeProduct);
    int GuessA = input.nextInt();
    int GuessB = input.nextInt();
    int GuessC = input.nextInt();
    int GuessSum = GuessA + GuessB + GuessC;
    int GuessProduct = GuessA * GuessB * GuessC;
    if (GuessProduct == CodeProduct && GuessSum == CodeSum)
    {
      return true;
    }
    else
    {
      return false;
    }
    
  }
  public static void main(String[] args) {

    int Difficulty = 1;
    int MaxDifficulty = 6;
        System.out.println("Hello student");

    while (Difficulty <= MaxDifficulty)
    {
      boolean bGameComplete = PlayGame(Difficulty);
      if (bGameComplete == true)
      {
        System.out.println("Nice Let's get Harder");
        ++Difficulty;
      }
      else
      {
        System.out.println("Oh no this won't do.");
      }
    }
    System.out.println("Great Job");
  }
}
